src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -312,8 +312,7 @@ public final void visitToken(DetailAST blockCommentNode) {
             // store as field, to share with child Checks
             context.get().blockCommentAst = blockCommentNode;
 
-            final LineColumn treeCacheKey = new LineColumn(blockCommentNode.getLineNo(),
-                    blockCommentNode.getColumnNo());
+            final LineColumn treeCacheKey = new LineColumn(blockCommentNode.getLineNo(), 0);
```

Comment:
this looks only because mostly javadoc comments are single on a line.
but this method does not demand this 
https://github.com/checkstyle/checkstyle/blob/22eb9f6eb5a2257b4b96f7cf3e45bae9efb6feaa/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtil.java#L141

so we potentially can have multiple javadocs comments on same line and we should not reuse wrong comment by taking it from cache.
I think we need to make test for this, regression will not find this as it will be very weird code, but as it is compile-able we have to support it.

Reply:
@Kevin222004 ,  do you remember in what PR we killed some mutation over this cache of javadocs ? this code looks very familiar and we dealt with it recently.

Reply:
@romani I have go through the previous pull request and as far as I can recall, we have not addressed any caching-related changes in the Javadocs.

We have only addressed caching in the Filters at https://github.com/checkstyle/checkstyle/commit/cde9c3e3de401c8202d80f0a4d18cae3532b2daa

Reply:
https://github.com/checkstyle/checkstyle/commit/75a99dceabb5bf8a4e92863f5e789aef57e8d3b9#diff-4bf1e51eabb8328d02e5fe62d69b7994c3490110ab1038e9f9df830496a561eeR188

This is where we dealt  with it.
https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocPosition.java

@suniti0804, please look at this Input and try to make new Input that has multiple javadoc on same line.
You might be able to archive this by having class with single method , both of them has javadoc comments, and you remove all end of lines to make it all single line.